src/uct/tcp/tcp_sockcm.c

```diff
@@ -103,9 +159,7 @@ UCS_CLASS_CLEANUP_FUNC(uct_tcp_sockcm_t)
     UCS_ASYNC_BLOCK(self->super.iface.worker->async);
 
     ucs_list_for_each_safe(sa_arg_ctx, tmp, &self->sa_arg_list, list) {
-        ucs_async_remove_handler(sa_arg_ctx->fd, 1);
-        close(sa_arg_ctx->fd);
-        ucs_free(sa_arg_ctx);
+        uct_tcp_close_sa_arg(sa_arg_ctx);
```

Comment:
`UCS_CLASS_INIT_FUNC` has to be updated to initialize `self->sa_arg_list`:
```
ucs_list_head_init(&self->sa_arg_list);
```

Reply:
it already does this - https://github.com/openucx/ucx/blob/master/src/uct/tcp/tcp_sockcm.c#L92